internal/server/presentation/log.go

```diff
@@ -59,6 +71,26 @@ func (h *AMQPLogHandler) HandleLog(msg amqp.Delivery) {
 	h.SendResponse(utils.OK, "OK", msg.ReplyTo, msg.CorrelationId)
 }
 
+// TODO: [Server] Improve the current RPC Implementation to reduct frontend delays
+// https://github.com/okuda-seminar/log_service/issues/85
+func (h *AMQPCTRLogHandler) HandleCTRLog(msg amqp.Delivery) error {
+	req, err := ParseAMQPCTRLog(msg)
+	if err != nil {
+		return fmt.Errorf("failed to parse CTR log request: %v", err)
```

Comment:
I mentioned using the `error` type, but let's control it with `nack` and `ack` instead.  
This way, messages will be requeued if the database goes down.

Reply:
```
msg.Nack(false, true)
```